AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -2315,31 +2335,53 @@ function New-ADAccountForStorageAccount {
             -Filter { ServicePrincipalNames -eq $spnValue } `
             -Server $Domain
 
-    if ($null -ne $computerSpnMatch -or $null -ne $userSpnMatch) {
-        Write-Error -Message "An AD object with a Service Principal Name of $spnValue already exists within AD. This might happen because you are rejoining a new storage account that shares names with an existing storage account, or if the domain join operation for a storage account failed in an incomplete state. Delete this AD object (or remove the SPN) to continue. See https://docs.microsoft.com/azure/storage/files/storage-troubleshoot-windows-file-connection-problems for more information." -ErrorAction Stop
+    if (($null -ne $computerSpnMatch) -and ($null -ne $userSpnMatch)) {
+        Write-Error -Message "There are already two AD objects with a Service Principal Name of $spnValue in domain $Domain." -ErrorAction Stop

```

Comment:
I think we should give better guidance here.  can we log the object names/distinguished names and tell them having two objects isn't supported.  and guidance to delete one of the objects?

Reply:
by the way, I'm assuming this is pretty rare so I don't think this should be blocking this PR